### PR TITLE
feat: br ci support run_if_changed

### PR DESCRIPTION
**Pull request by purelind**
The br & lightning pipeline supports triggering based on conditions. If code changes are detected in the corresponding directory, the corresponding test pipeline will be triggered. Block PR merging if the test pipeline execution fails.

It should be noted that because prow does not support configuring these two parameters at the same time (`run_if_changed` and `skip_if_only_changed`), after applying this change. If there are changes in pure document categories for br & lightning, it will also trigger the test pipeline.

### CODE REVIEW

**Comment by purelind** [comment]
/hold

Wait for the RD of  br & lightning to review and merge into effect when CI is idle.

**Comment by wuhuizuo** [comment]
/lgtm
/approve cancel

wait for more LGTM from `br` and `lightning` teams.

**Review by BornChanger — APPROVED** [review]

**Review by wuhuizuo — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
